aten/src/ATen/native/Activation.cpp

```diff
@@ -43,6 +44,218 @@ Tensor & rrelu_(Tensor & self, Scalar lower, Scalar upper, bool training, Genera
   return at::rrelu_with_noise_(self, self.type().tensor(), lower, upper, training, generator);
 }
 
+// -----------------------------------
+// prelu forward
+// -----------------------------------
+template <typename scalar_t>
+void inline prelu_cpu_kernel_multi_weights(
+  Tensor& result,
+  const Tensor& input,
+  const Tensor& weight,
+  int64_t input_stride0,
+  int64_t input_stride1) {
+
+  int64_t i;
+  int64_t input_numel = input.numel();
+  auto result_data = result.data<scalar_t>();
+  auto input_data = input.data<scalar_t>();
+  auto weight_data = weight.data<scalar_t>();
+
+  // multiply values at each channel with weight[channel]
+  #pragma omp parallel for private(i) if (input_numel > 1000)
+  for (i = 0; i < input_numel; i++) {
+    int64_t channel = (i % input_stride0) / input_stride1;
```

Comment:
This isn't similar to the original implementation at all. Indeed as @resistor said, modulo and division are expensive. It is possible that the original simple strategy of parallelizing over batch dim is better.

Reply:
I suspect the best combination is to parallelize over the outermost dim and vectorize over the innermost dim.

Reply:
@cpuhrsch makes a good point that it's not clearly worth the effort to figure out the inner vectorization for this inline implementation.